### ISSUE TITLE
Fan Speed

### ISSUE THREAD

**Issue by agamboagg** [0x4a616e/homebridge-smartthings-ac#4]
Hi, first of all, thank you very much and excellent Plug in!
I would like to ask if it is possible to control the fan speed. Currently in my AC only the temperature can be moved, but there is no way to change the fan speed.

Additionally, if the app could read the current speed and not move it when adjusting the temperature, that would be great.
Currently when I change the temperature it changes the fan speed to Auto, it even changes the speed to Auto the moment I enter the AC in the Home application. (I usually use my AC on low speed or Windfree).

One last request, do you think that later on, you can put a switch to turn the Windfree mode on and off?

Thank you!

**Comment by 0x4a616e** [comment]
Hi @agamboagg! I'd love to see fan speed controls as well. I already started with an implementation but there is one small logical challenge: Homebridge uses a percentage value for the fan speed whereas SmartThings uses discrete values `off`, `low`, `medium`, ` high`, `max` and `auto`. While I can somehow map most values to fractions and vice versa, it is not clear how to handle `auto`... The key question is what value to show if the fan speed is set to `auto` and how to set the speed to `auto` using homebridge controls. Unfortunately, you cannot add arbitrary controls to a heater cooler device. This is all what is supported: https://developers.homebridge.io/#/service/HeaterCooler
If you have a good idea, I'd be happy to implement it!

````

**Comment by agamboagg** [comment]
Hi, thank you very much for the reply!
The way I would put it, although it may seem a bit counterintuitive, is:

0% -> off
20% -> low
40% -> mid
60% -> high
80% -> max
100% -> Auto

However, could it be made to not change the fan speed when the temperature is changed? I mean that even if you don't allow the fan speed to be changed, read the speed at which it is set, and when the temperature is changed, keep that speed.

Example: the AC is at 23c Mid speed, and I want to set it to 22c, which reads that it is at Mid speed and change it to 22c Mid, without having to give the option to adjust the fan.

One question, can you create a switch external to the AC to modify some parameter? in this case it would be to make an independent switch in Homebridge to the Heater / Cooler to activate the Auto speed of the AC ...

If I don't explain myself, please tell me, since my native language is Spanish.

Greetings!

**Comment by beagleboys** [comment]
Can auto be set as a separate switch? when on you can't change the speed settings, when off you can change the fan speed slider

**Comment by 0x4a616e** [comment]
Sorry for being a bit unresponsive during the last days but I was on vacation with limited internet access 😉 So first of all, @agamboagg i thing what your suggesting is perfectly clear. I had a look at other plugins and it seems they are using a similar approach: https://www.npmjs.com/package/homebridge-gree-heatercooler-v2 uses 100% as `auto`, https://www.npmjs.com/package/homebridge-tado-ac does it for 0%.

I'm still haven't completely made up my mind as one the one hand, I'd like to keep as much functionality as possible within one HomeKit accessory. On the other hand, using 0% or 100% as auto is not very intuitive.

Currently, I have a tendency to an additional switch as @beagleboys suggested and remove the fan speed control from the AC as long as `auto` is enabled. I guess I'll just give it a try and see how it feels.

**Comment by cicciovo** [comment]
Any update? 😅

**Comment by agamboagg** [comment]
> Any update? 😅

x2 😜, me too

**Comment by 0x4a616e** [comment]
Yeah, thanks for reminding me! it's freaking cold in Germany right now, so I'm currently not using my AC at all... so yeah, this somehow disappeared from my radar 😬 I'll see what I can do within the next weeks!

**Comment by agamboagg** [comment]
> Yeah, thanks for reminding me! it's freaking cold in Germany right now, so I'm currently not using my AC at all... so yeah, this somehow disappeared from my radar 😬 I'll see what I can do within the next weeks!

Thank you from 35*C Mexico!

**Comment by 0x4a616e** [comment]
I gave it a try here https://github.com/0x4a616e/homebridge-smartthings-ac/tree/set_fan_speed but for some reason, it doesn't work. It looks like samsung is ignoring the fan speed command... The command completes with status `success` but nothing happens... Any ideas maybe? This may be unrelated, but i also cannot find the fan speed in the smartthings app anymore.

**Comment by agamboagg** [comment]
> I gave it a try here https://github.com/0x4a616e/homebridge-smartthings-ac/tree/set_fan_speed but for some reason, it doesn't work. It looks like samsung is ignoring the set fan speed command... The command completes with status `success` but nothing happens... Any ideas maybe? This may be unrelated, but i also cannot find the fan speed in the smartthings app anymore.

Hi, I'm using HOOBS, so it won't let me install it from the branch, I could only install it from the main one, could you upload it to the main branch momentarily for me to install it?
